File: samples/app/adapters/env_client.js
/**
 * ENVClient implementation
 *
 * Implements the ENVClient Required Port.
 * Provides access to environment variables.
 */

class ENVAdapter {
  /**
   * Get environment variable
   * @param {string} key - The environment variable name
   * @returns {string|null} The value or null if not found
   */
  get(key) {
    return process.env[key] ?? null;
  }

  /**
   * Check if environment variable exists
   * @param {string} key - The environment variable name
   * @returns {boolean} True if exists
   */
  has(key) {
    return key in process.env;
  }

  /**
   * Get all environment variables
   * @returns {Object} All environment variables
   */
  getAll() {
    return { ...process.env };
  }
}

module.exports = ENVAdapter;

File: samples/adapters/env_client.rs
/// ENVClient implementation
///
/// Implements the ENVClient Required Port.
/// Provides access to environment variables.

use state_engine::ports::required::ENVClient;
use std::collections::HashMap;

pub struct ENVAdapter;

impl ENVAdapter {
    pub fn new() -> Self {
        Self
    }

    /// Check if environment variable exists
    pub fn has(&self, key: &str) -> bool {
        std::env::var(key).is_ok()
    }

    /// Get all environment variables
    pub fn get_all(&self) -> HashMap<String, String> {
        std::env::vars().collect()
    }
}

impl ENVClient for ENVAdapter {
    fn get(&self, key: &str) -> Option<String> {
        std::env::var(key).ok()
    }
}
